src/contents/key_pair.rs

```diff
@@ -1,48 +1,68 @@
 use super::encryption::unseal_box;
 use super::public_key_info::{KeyType, PublicKeyInfo};
-use secp256k1::{Message, Secp256k1, SecretKey};
+use k256::{
+    ecdsa::{
+        SigningKey,
+        VerifyKey,
+        Signature,
+        signature::Signer,
+        recoverable,
+    },
+};
 use serde::{Deserialize, Serialize};
-use ursa::{
-    encryption::symm::prelude::*,
-    kex::x25519::X25519Sha256,
-    kex::KeyExchangeScheme,
-    keys::{KeyGenOption, PrivateKey},
-    signatures::prelude::*,
+use x25519_dalek::{
+    x25519,
+    PublicKey,
+    StaticSecret,
+};
+use crypto_box::SecretKey;
+use ed25519_dalek::{
+    self,
+    Keypair,
 };
+use rand_core::OsRng;
 use crate::Error;
-use sha3::{Digest, Keccak256};
 
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct KeyPair {
     #[serde(flatten)]
     pub public_key: PublicKeyInfo,
-    pub private_key: PrivateKey,
+    pub private_key: Vec<u8>,
 }
 
 impl KeyPair {
-    pub fn new(key_type: KeyType, priv_key: &Vec<u8>) -> Result<KeyPair, Error> {
+    pub fn new(key_type: KeyType, priv_key: &Vec<u8>) -> Result<Self, Error> {
         let (pk, sk) = match key_type {
             KeyType::Ed25519VerificationKey2018 => {
-                Ed25519Sha512::expand_keypair(&priv_key).map_err(|e| Error::UrsaCryptoError(e))?
-            }
+                let sk = ed25519_dalek::SecretKey::from_bytes(priv_key)
+                    .map_err(|e| Error::EdCryptoError(e))?;
+                let pk: ed25519_dalek::PublicKey = (&sk).into();
+                let mut pk = *pk.as_bytes();
+                let mut sk = *sk.as_bytes();
+                (pk.iter_mut().map(|i| return *i).collect::<Vec<u8>>(),
```

Comment:
is there an alternative way to get the types we want here? or if this is the most convenient way, can we do a `impl From<K>`? we should collab on the type abstraction we want to use for key pairs

Reply:
done (where possible. had to leave `.map().collect()` in one place...) =)